src/xc_write_output.F

```diff
@@ -48,7 +48,7 @@ SUBROUTINE xc_write(output_unit, xc_section, lsd)
 
       CHARACTER(LEN=2*default_string_length)             :: shortform
       CHARACTER(LEN=20)                                  :: tmpStr
-      CHARACTER(LEN=20*default_string_length)            :: reference
+      CHARACTER(LEN=200*default_string_length)           :: reference
```

Comment:
is there a way we can inspect the length needed at runtime and use `ALLOCATE(...)` instead?

Reply:
I can only use a smaller value for the length as it was just a number which worked (20*default_string_length does not). Our reference of a LibXC functional may have up to 5 different references, a DOI, external parameters with description and scaling factor. So theoretically, the string can be very long.

Reply:
There should be a safeguard in [xc_libxc_wrap_info_refs](https://github.com/cp2k/cp2k/blob/master/src/xc/xc_libxc_wrap.F#L138) that prevents it from writting out of bounds. Then we can pass in a reasonably sized buffer, e.g. `10*default_string_length`. I think, 10 lines per functional ought to be enough.

Reply:
I have added a variable length reference character and a safe-guard.